### PR TITLE
feat: support unnamed default export

### PR DESCRIPTION
**Pull request by usualoma**
What about making it available for unnamed functions?

### CODE REVIEW

**Comment by yusukebe** [comment]
Hey @usualoma 

This is great! We should support as many ways to `export` as possible.

I think we can merge this PR right now, but we can consider what @aulneau said:

> I've also noticed that islands specifically need export default function to work. I can't do a named export or a separate declaration and default export. Is that expected?

**Comment by usualoma** [comment]
Hi @yusukebe 
Thank you for reviewing

> I've also noticed that islands specifically need export default function to work. I can't do a named export or a separate declaration and default export. Is that expected?

This PR doesn't resolve the points made in this comment.
* named export
* separate declaration and default export

I think we can do this by adding https://github.com/usualoma/honox/pull/1 changes to this. Do you want to include this?

**Comment by usualoma** [comment]
Oh, we also need to change client.ts. for https://github.com/usualoma/honox/pull/1

**Comment by usualoma** [comment]
Corrections and refactoring have been completed. If you look at the test cases, you will understand how things have changed.
How about this?
https://github.com/usualoma/honox/pull/1

**Comment by yusukebe** [comment]
@usualoma 

Thanks for creating another one!

> I think it is easy to understand that "only default export is supported".

I fully agree with this. I think it would be easy to explain to users. So I would like you to add only "*separate declaration and default export*" to this PR.

**Comment by usualoma** [comment]
@yusukebe Okay, I want to refactor it, so wait a minute.

**Comment by usualoma** [comment]
Done. We can now write in the following styles

```ts
export default function() {
  return <h1>Hello</h1>
}
```

```ts
export default () => {
  return <h1>Hello</h1>
}
```

```ts
const ExportViaVariable = () => <h1>Hello</h1>
export default ExportViaVariable
```

```ts
const ExportViaVariable = () => <h1>Hello</h1>
export { ExportViaVariable as default }
```

File: test/unit/vite/island-components.test.ts [hunk 1]
```diff
@@ -30,10 +30,80 @@ export default WrappedBadge;`
...
+    expect(result).toBe(
+      `const WrappedExportViaVariable = function (props) {
+  return import.meta.env.SSR ? <honox-island component-name="ExportViaVariable.tsx" data-serialized-props={JSON.stringify(Object.fromEntries(Object.entries(props).filter(([key]) => key !== "children")))}><ExportViaVariable {...props}></ExportViaVariable>{props.children ? <template data-hono-template="">{props.children}</template> : null}</honox-island> : <ExportViaVariable {...props}></ExportViaVariable>;
+};
+export default WrappedExportViaVariable;`
+    )
+  })
+
+  it('export via specifier', () => {
+    const code = `const utilityFn = () => {}
```
Comment (yusukebe) [inline test/unit/vite/island-components.test.ts:96]: It's good to test the utility function too!

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@usualoma Thanks! Merge now.

**Pull request merged**